js/app.js

```diff
@@ -11,128 +11,128 @@ Calculator.prototype.getHistoryAsString = function() {
     return this.history.join('\n');
 }
 
-// Calculator.prototype.add = function(num1, num2) {
-//     // 1. zamień wartości przekazane przez parametr na typ number
-//     // 2. sprawdź czy są one poprawne
-//     // 3. jeśli tak to wykonaj działanie i zapisz jego resultat
-//     // 4. dodaj do historii operacji to działanie w fomie: 1 + 1 = 2
+Calculator.prototype.add = function(num1, num2) {
+    // 1. zamień wartości przekazane przez parametr na typ number
+    // 2. sprawdź czy są one poprawne
+    // 3. jeśli tak to wykonaj działanie i zapisz jego resultat
+    // 4. dodaj do historii operacji to działanie w fomie: 1 + 1 = 2
 
-//     const number1 = parseFloat(num1);
-//     const number2 = parseFloat(num2);
-
-//     if (isNaN(number1) || isNaN(number2)) {
-//         alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
-//         location.reload();
-//     }
-
-//     const result = number1 + number2;
-//     const operation = `${number1} + ${number2} = ${result}`;
-//     this.history.push(operation);
-// }
-
-// Calculator.prototype.subtract = function(num1, num2) {
-
-//     const number1 = parseFloat(num1);
-//     const number2 = parseFloat(num2);
+    const number1 = parseFloat(num1);
+    const number2 = parseFloat(num2);
 
-//     if (isNaN(number1) || isNaN(number2)) {
-//         alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
-//         location.reload();
-//     }
+    if (isNaN(number1) || isNaN(number2)) {
+        alert('Podana wartość musi być liczbą! Wprowadź poprawne dane!');
+        location.reload();
+    }
```

Comment:
Zabrakło mi przerzucenie tej cześci kodu z każdej funkcji do linii 166 czyli mamy najpierw konwersję do `number`, potem sprawdzamy czy są to poprawne wartości, a dopiero potem uruchamiamy działanie.
Dzięki temu nie powtarzamy operacji sprawdzania danych za każdym razem, tylko raz.